website/packages/docs/src/pages/pkg-babel-plugin.mdx

```diff
@@ -41,41 +41,118 @@ Configure in your Babel config.
+#### classNameCompressionMap
 
-### optimizeCss: boolean
+Don't use this!
 
-Will run additional cssnano plugins to normalize CSS during build.
+An internal and experimental option used to compress class names.
 
-Defaults to `true`.
+Note that in `@compiled/webpack-loader` and `@compiled/parcel-config`, `classNameCompressionMap` requires `extract` to be true as well.
 
-### onIncludedFiles: (files: string[]) => void
+#### extensions
 
-Will callback at the end of a file pass with all imported files that were statically evaluated into the file.
+Extensions that we should consider code. We use these to identify if a file should be parsed.
 
-### addComponentName: boolean
+- Type: `string[]`
+- Default: `['.js', '.jsx', '.ts', '.tsx']`
 
-Add the component name as class name to DOM in non-production environment if styled is used.
+#### importReact
+
+Will import React into the module if it is not found.
+When using `@babel/preset-react` with the [automatic runtime](https://babeljs.io/docs/en/babel-preset-react#react-automatic-runtime) this is not needed and can be set to `false`.
 
-Default to `false`
+- Type: `boolean`
+- Default: `true`
 
-### importSources: string[]
+#### importSources
 
 Additional libraries that should be parsed as though they were `@compiled/react` imports.
 
 Specifying this is important if you are using Compiled APIs through another package using the `createStrictAPI` function.
+
+- Type: `string[]`
+- Default: `undefined`
+
+#### increaseSpecificity
+
+An experimental option that increases the specificity of all generated Compiled classes. Don't use this!
+
+#### nonce
+
+[Security nonce](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce) that will be applied to inline style elements if defined.
+
+- Type: `string`
+- Default: `false`
+
+#### onIncludedFiles
+
+An internal option. Callback fired at the end of the file pass when files have been included in the transformation.
+
+- Type: `(files: string[]) => void`
+- Default: `undefined`
+
+#### optimizeCss
+
+Will run additional `cssnano` plugins to normalize CSS during the build.
+
+- Type: `boolean`
+- Default: `true`
+
+#### parserBabelPlugins
+
+Babel parser plugins to be used when parsing the source code. Define these to enable extra babel parsers (for example, typescript).
+See the [babel docs](https://babeljs.io/docs/en/plugins/#syntax-plugins) for more context.
+
+Usually, you will set this option through [`@compiled/parcel-config`] or [`@compiled/webpack-loader`](/webpack-loader) -- see there for examples.
+
+#### processXcss
+
+An option we are only using internally. Don't use!
+
+#### resolver
+
+A custom resolver used to statically evaluate import declarations, specified as either an object or module path. If this is an object, it should contain a `resolveSync` function with the following type signature: `(context: string, request: string) => string`.
+
+- Type: `string | Resolver`
+- Default: `undefined`
+
+#### sortAtRules
+
+Unused.
```

Comment:
I reckon dealing with stuff that's unused, deprecated, obsolete etc would be out of scope of this PR - as a website/docs-focussed PR, I want to avoid making changes to the Compiled packages

I've made an issue so that we can properly clean these up in the future https://github.com/atlassian-labs/compiled/issues/1690